### ISSUE TITLE
Parent relation not filtering properly when using hashed id

### ISSUE THREAD

**Issue by kchapple** [wp-graphql/wp-graphql#2305]
I was having an issue querying a parent/child using the 'id' field (hashed id) on a custom post type. I was also able to reproduce the error with posts and revisions, with no other plugins activated using WP GraphQL Version 1.7.2. 

Running the following queries I would expect GetRevisions and GetRevisionsHash to return the same results. Instead, GetRevisions returns the expected 2 results, while GetRevisionsHash returns many more, perhaps all, results.

To reproduce, create a couple posts with a couple revisions of each post..

Then, run the GetPost query to get the hashed ID and the databaseId. 

Then run GetRevisions using database ID for parent and see expected results.

Then run GetRevisionsHash using id (hash) for parent and see incorrect results.

```
query GetPost {
  post(id: "204129", idType: DATABASE_ID) {
    id
    postId
    title
    content
  }
}

query GetRevisions {
  revisions(where: {parent: "204129"}) {
    nodes {
      ... on Post {
        id
        databaseId
        title
      }
    }
  }
}

query GetRevisionsHash {
  revisions(where: {parent: "cG9zdDoyMDQxMzE="}) {
    nodes {
      ... on Post {
        id
        databaseId
        title
      }
    }
  }
}
```

**Comment by justlevine** [comment]
@kchapple Unless I am misunderstanding the issue, this is expected behavior and _all_ where args throughout core schema require the use of databaseId, not the global id.

**Comment by bhardie** [comment]
Related to #997

**Comment by justlevine** [comment]
@bhardie you sure this would be breaking? the `parent` arg already takes an `ID`, so if WPGraphQL was to implement this (not saying it necessarily should), we could just just run `ID` types through a parser function to get the database ID.

Something along the lines of:

```php
$database_id = get_database_id( $where['global_or_db_id'] );
...
function get_database_id( $id ) : int {
  if( is_numeric( $id ) {
    return (int) $id;
  }

  $id_parts = Relay::fromGlobalId( $id );
  
  if ( ! isset ( $id_parts['id'] ) ){
   // Handle the error
  }
  
  return (int) $id_parts['id'];
}

**Comment by jasonbahl** [comment]
I think the ultimate goal, and why @bhardie labeled it breaking, is what's described in #997 

Ideally, any field that is asking for an ID (in this case the `post( id )` and the `where: { parent }` fields) should be of type ID and accept either a Global ID or a databaseId. 

#2322 gets us a step in the right direction toward getting the inconsistent behavior resolved, but some fields are still of the `Int` Scalar type and will not accept a Global ID as input until changed to `ID` Type. 

Also, there are probably cases where accepting _just_ a databaseId might not have enough information. 

Like a query such as: `node( id: $id ) { ... }` if the ID were a database ID, that's not enough information to resolve, as a `node` might be stored in different tables. We could have a Term with ID 1, a User with id 1, a Post with id 1. The database ID alone is not enough information to know how to resolve the thing. 

So, we need to consider this as we move forward. 

Does the field have enough context to allow a databaseId as input or not?

**Comment by justlevine** [comment]
The nuance I'm making is that while #997 _is_ breaking, #998 is *not*. If an input type is already `ID` (like `RootQueryToContentRevisionUnionConnectionWhereArgs.parent`), then its not a breaking change for that field to handle both database and global ids.

And considering #1095 and the consistent reports of users trying to use a global database ID  in a `ID` type input across the schema (like this one), I think the short-term path forward is to work on #998 (changing input descriptions to specify when an `ID` takes _only_ a global id), and then handle converting `Int` inputs to `ID`s when we're ready for a breaking release.

**Comment by justlevine** [comment]
Leaving closed in favor of #998
